### PR TITLE
chore(): edit icon only on own beams

### PR DESCRIPTION
**Pull request by themonster2015**
## Description

* Display edit icon only on own beam cards.

## Checklist

- [ ] I have read the **README** document
- [ ] I have read the **CONTRIBUTING** document
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed
- [ ] My commit message follows [Conventional Commit Guideline](https://conventionalcommits.org/)

### CODE REVIEW

File: ui/design-system-components/src/components/Entry/EntryCard/index.tsx [hunk 1]
```diff
@@ -146,13 +146,11 @@ const EntryCard: React.FC<EntryCardProps> = props => {
...
           {entryData?.createdAt &&
-            (!editable && notEditableLabel ? (
+            (editable ? (
               <Tooltip placement={'top'} content={notEditableLabel}>
                 {editableIconButtonUi}
               </Tooltip>
-            ) : (
-              editableIconButtonUi
-            ))}
+            ) : null)}
```
Comment (didd) [inline ui/design-system-components/src/components/Entry/EntryCard/index.tsx]: That logic was there for a reason :) Can you share the reason why removed it ? I agree `!editable` check isn't needed but the rest are important.

The logic was there for two reasons. One, when `notEditableLabel` is set then it adds unnecessary tooltip element into a page and two when you hover on the `edit` icon then you will see unwanted arrow.

image

Comment (themonster2015) [reply ui/design-system-components/src/components/Entry/EntryCard/index.tsx]: thanks for pointing it out Didd :). I overlooked the notEditableLabel prop. Will take a closer look.

File: ui/apps/lists/src/components/lists-page.tsx [hunk 2]
```diff
@@ -21,15 +21,16 @@ const ListsPage: React.FC<unknown> = () => {
 
   const bookmarkDelete = null;
 
+  const { isLoggedIn } = useLoggedIn();
+
   const profileDataReq = useGetMyProfileQuery(null, {
```
Comment (didd) [inline ui/apps/lists/src/components/lists-page.tsx]: We can use `loggedProfileData` from `useLoggedIn` hook instead of recomputing it here :)

Comment (themonster2015) [reply ui/apps/lists/src/components/lists-page.tsx]: This component needs updating yes

File: ui/lib/feed/src/components/cards/beam-card.tsx [hunk 3]
```diff
@@ -22,10 +22,11 @@ type BeamCardProps = Pick<
   | 'disableActions'
 > & {
   entryData: AkashaBeam;
+  loggedProfileData: Profile;
```
Comment (didd) [inline ui/lib/feed/src/components/cards/beam-card.tsx:25]: This prop isn't needed. We can use the `isViewer` flag from the author object to decide whether to show or hide the edit button. As a result you don't need to set the editable prop for such purpose. 

Btw there is a new design I have been working on that of placing the edit button within an ellipsis menu and I have taken care of all this on the PR #2086.

**Pull request closed without merging**